### ISSUE TITLE
Evaluate whether we need the set_union and set_difference methods

### ISSUE THREAD

**Issue by nand4011** [momentohq/client-sdk-rust#169]
None of the other SDKs expose union and difference directly to the user. Can they be removed and replaced by other set methods?

**Comment by cprice404** [comment]
yes, we should definitely remove them and make the API consistent with the other SDKs.

**Comment by nand4011** [comment]
We will delete them when we get rid of the old cache client.
